reference-architecture/azure-ansible/3.6/allinone.sh

```diff
@@ -431,6 +431,213 @@ cat <<EOF > /home/${AUSERNAME}/postinstall.yml
+  - name: Stage 4 - X509
+    command: "openssl x509 -req -passin pass:{{idm_xpassword}} -CA {{idm_dir}}/xpaas.crt -CAkey {{idm_dir}}/xpaas.key -in {{idm_dir}}/sso.csr -out {{idm_dir}}/sso.crt -days 365 -CAcreateserial"
+  - name: Stage 5 - IMPORT CRT
+    command: "keytool  -noprompt -deststorepass {{idm_xpassword}} -import -file {{idm_dir}}/xpaas.crt  -storepass {{idm_xpassword}} -keypass {{idm_xpassword}} -alias xpaas.ca -keystore {{idm_dir}}/sso-https.jks"
+  - name: Stage 6 - IMPORT SSO
+    command: "keytool  -noprompt -deststorepass {{idm_xpassword}} -storepass {{idm_xpassword}} -keypass {{idm_xpassword}}  -import -file {{idm_dir}}/sso.crt -alias sso-https-key -keystore {{idm_dir}}/sso-https.jks"
+  - name: Stage 7 - IMPORT XPAAS
+    command: "keytool -noprompt -deststorepass {{idm_xpassword}} -storepass {{idm_xpassword}} -keypass {{idm_xpassword}}   -import -file {{idm_dir}}/xpaas.crt -alias xpaas.ca -keystore {{idm_dir}}/truststore.jks"
+  - name: Stage 8 - GENSECKEY
+    command: "keytool  -deststorepass {{idm_xpassword}} -storepass {{idm_xpassword}} -keypass {{idm_xpassword}} -genseckey -alias jgroups -storetype JCEKS -keystore {{idm_dir}}/jgroups.jceks"
+  - name: Stage 9 - OCCREATE SECRET
+    command: "oc create secret generic sso-app-secret --from-file={{idm_dir}}/jgroups.jceks --from-file={{idm_dir}}/sso-https.jks --from-file={{idm_dir}}/truststore.jks"
+  - name: Stage 10 - OCCREATE SECRET ADD
+    command: "oc secret add sa/{{sso_project}}-service-account secret/sso-app-secret"
+  - name: Stage 11 - Create App Parameters
+    blockinfile:
+       path: "{{idm_dir}}/sso.params"
+       create: yes
+       block: |
+         HOSTNAME_HTTP="nlogin.{{sso_domain}}"
+         HOSTNAME_HTTPS="login.{{sso_domain}}"
+         APPLICATION_NAME="{{sso_project}}"
+         HTTPS_KEYSTORE="sso-https.jks"
+         HTTPS_PASSWORD="{{idm_xpassword}}"
+         HTTPS_SECRET="sso-app-secret"
+         JGROUPS_ENCRYPT_KEYSTORE="jgroups.jceks"
+         JGROUPS_ENCRYPT_PASSWORD="{{idm_xpassword}}"
+         JGROUPS_ENCRYPT_SECRET="sso-app-secret"
+         SERVICE_ACCOUNT_NAME={{sso_project}}-service-account
+         SSO_REALM=cloud
+         SSO_SERVICE_USERNAME="{{sso_username}}"
+         SSO_SERVICE_PASSWORD="{{sso_password}}"
+         SSO_ADMIN_USERNAME=admin
+         SSO_ADMIN_PASSWORD="{{sso_password}}"
+         SSO_TRUSTSTORE=truststore.jks
+         SSO_TRUSTSTORE_PASSWORD="{{idm_xpassword}}"
+
+  - name: Stage 10 - OCCREATE SECRET ADD
+    command: oc new-app sso71-postgresql --param-file {{idm_dir}}/sso.params -l app=sso71-postgresql -l application=sso -l template=sso71-https
+  - set_fact: sso_token_url="https://login.{{sso_domain}}/auth/realms/cloud/protocol/openid-connect/token"
+  - name: Pause for app create
+    pause:
+      minutes: 4
```

Comment:
I'd say to wait until the pod is running instead just waiting a number of minutes

Reply:
Actually I found a bug in the java side, I was doing a rest call to see if alive, and found it hosed to sso process and it would never work, so the sleep followed by the retry logic fixes this.